pydantic/type_adapter.py

```diff
@@ -233,100 +208,122 @@ def __init__(
+            self.validator = create_schema_validator(
+                schema=self.core_schema,
                 schema_type=self._type,
                 schema_type_module=self._module_name,
                 schema_type_name=str(self._type),
                 schema_kind='TypeAdapter',
                 config=core_config,
                 plugin_settings=config_wrapper.plugin_settings,
             )
-            self._serializer = SchemaSerializer(self._core_schema, core_config)
-
-        if rebuild_mocks and isinstance(self._core_schema, _mock_val_ser.MockCoreSchema):
-            self._core_schema.rebuild()
-            self._init_core_attrs(rebuild_mocks=False)
-            assert not isinstance(self._core_schema, _mock_val_ser.MockCoreSchema)
-            assert not isinstance(self._validator, _mock_val_ser.MockValSer)
-            assert not isinstance(self._serializer, _mock_val_ser.MockValSer)
-
-    @cached_property
-    @_frame_depth(2)  # +2 for @cached_property and core_schema(self)
-    def core_schema(self) -> CoreSchema:
-        """The pydantic-core schema used to build the SchemaValidator and SchemaSerializer."""
-        if self._core_schema is None or isinstance(self._core_schema, _mock_val_ser.MockCoreSchema):
-            self._init_core_attrs(rebuild_mocks=True)  # Do not expose MockCoreSchema from public function
-        assert self._core_schema is not None and not isinstance(self._core_schema, _mock_val_ser.MockCoreSchema)
-        return self._core_schema
-
-    @cached_property
-    @_frame_depth(2)  # +2 for @cached_property + validator(self)
-    def validator(self) -> SchemaValidator | PluggableSchemaValidator:
-        """The pydantic-core SchemaValidator used to validate instances of the model."""
-        if not isinstance(self._validator, (SchemaValidator, PluggableSchemaValidator)):
-            self._init_core_attrs(rebuild_mocks=True)  # Do not expose MockValSer from public function
-        assert isinstance(self._validator, (SchemaValidator, PluggableSchemaValidator))
-        return self._validator
-
-    @cached_property
-    @_frame_depth(2)  # +2 for @cached_property + serializer(self)
-    def serializer(self) -> SchemaSerializer:
-        """The pydantic-core SchemaSerializer used to dump instances of the model."""
-        if not isinstance(self._serializer, SchemaSerializer):
-            self._init_core_attrs(rebuild_mocks=True)  # Do not expose MockValSer from public function
-        assert isinstance(self._serializer, SchemaSerializer)
-        return self._serializer
+            self.serializer = SchemaSerializer(self.core_schema, core_config)
+
+        # TODO: I think we should move this to the rebuild pattern?
+        if isinstance(self.core_schema, _mock_val_ser.MockCoreSchema):
+            self.core_schema = self.core_schema.rebuild()  # type: ignore[assignment]
+            self._init_core_attrs(ns_resolver=ns_resolver, force=True)
```

Comment:
You're right. Looking into a fix now.

Reply:
Good question, looking into this.

Reply:
So, this happens when we have a model that deferred (or failed) schema building.

I think this logic needs to be more complex - should we reassign `__pydantic_core_schema__`? What about `__pydantic_validator__` and `__pydantic_serializer__`?

Reply:
@sydney-runkle was there some test against the duplicated schema building issue? To test the the GenerateSchema is called only ones in this scenario, in case there was a bug here?